lib/iris/cube.py

```diff
@@ -1658,7 +1658,7 @@ def fill_value(self, fill_value):
             # Convert the given value to the dtype of the cube.
             fill_value = np.asarray([fill_value])[0]
             target_dtype = self.dtype
-            if fill_value.dtype.kind == 'f' and target_dtype.kind == 'i':
+            if fill_value.dtype.kind == 'f' and target_dtype.kind in 'biu':
                 # Perform rounding when converting floats to ints.
                 fill_value = np.rint(fill_value)
```

Comment:
It rounds to the nearest integer and treats `0` as `False` and non-zero as `True` ... which is fair enough.